modules/scylladb/scylladb_test.go

```diff
@@ -0,0 +1,194 @@
+		require.NoError(t, err)
+	})
+
+	t.Run("test with shard awareness", func(t *testing.T) {
+		host, err := ctr.ConnectionHost(ctx, 19042)
+		require.NoError(t, err)
+
+		cluster := gocql.NewCluster(host)
+		session, err := cluster.CreateSession()
+		require.NoError(t, err)
+		defer session.Close()
+
+		var address string
+		err = session.Query("SELECT address FROM system.clients").Scan(&address)
+		require.NoError(t, err)
+	})
+}
+
+func TestScyllaWithConfigFile(t *testing.T) {
+	ctx := context.Background()
+
+	ctr, err := scylladb.Run(ctx,
+		"scylladb/scylla:6.2",
+		scylladb.WithConfigFile(filepath.Join("testdata", "scylla.yaml")),
+		scylladb.WithShardAwareness(),
+	)
+	require.NoError(t, err)
+	testcontainers.CleanupContainer(t, ctr)
+
+	t.Run("test without shard awareness", func(t *testing.T) {
+		host, err := ctr.ConnectionHost(ctx, 9042)
+		require.NoError(t, err)
+
+		cluster := gocql.NewCluster(host)
+		session, err := cluster.CreateSession()
+		require.NoError(t, err)
+		defer session.Close()
+
+		var cluster_name string
+		err = session.Query("SELECT cluster_name FROM system.local").Scan(&cluster_name)
+		require.NoError(t, err)
+
+		require.Equal(t, "Amazing ScyllaDB Test", cluster_name)
+	})
+
+	t.Run("test with shard awareness", func(t *testing.T) {
+		host, err := ctr.ConnectionHost(ctx, 19042)
+		require.NoError(t, err)
+
+		cluster := gocql.NewCluster(host)
+		session, err := cluster.CreateSession()
+		require.NoError(t, err)
+		defer session.Close()
+
+		var cluster_name string
+		err = session.Query("SELECT cluster_name FROM system.local").Scan(&cluster_name)
+		require.NoError(t, err)
+
+		require.Equal(t, "Amazing ScyllaDB Test", cluster_name)
+	})
+}
+
+func TestScyllaWithAlternator(t *testing.T) {
+	ctx := context.Background()
+
+	alternatorPort := 8000
+	ctr, err := scylladb.Run(ctx,
+		"scylladb/scylla:6.2",
+		scylladb.WithAlternator(alternatorPort),
+	)
+	require.NoError(t, err)
+	testcontainers.CleanupContainer(t, ctr)
+
+	client, err := getDynamoAlternatorClient(t, ctr, alternatorPort)
+	require.NoError(t, err)
+	err = createTable(client)
+	require.NoError(t, err)
+}
+
+func TestScyllaWithoutAlternator(t *testing.T) {
+	ctx := context.Background()
+
+	alternatorPort := 8000
+	ctr, err := scylladb.Run(ctx,
+		"scylladb/scylla:6.2",
+	)
+	require.NoError(t, err)
+	testcontainers.CleanupContainer(t, ctr)
+
+	client, err := getDynamoAlternatorClient(t, ctr, alternatorPort)
+	require.Error(t, err)
+	err = createTable(client)
+	require.Error(t, err)
+}
+
+type scyllaAlternatorResolver struct {
+	HostPort string
+}
+
+func (r *scyllaAlternatorResolver) ResolveEndpoint(ctx context.Context, params dynamodb.EndpointParameters) (smithyendpoints.Endpoint, error) {
+	return smithyendpoints.Endpoint{
+		URI: url.URL{Host: r.HostPort, Scheme: "http"},
+	}, nil
+}
+
+func getDynamoAlternatorClient(t *testing.T, c *scylladb.Container, port int) (*dynamodb.Client, error) {
```

Comment:
suggestion: if the assertions in the calling tests are all the same (create a table using the client), you may want to merge them into this helper function, and name it `assertCreateTable(t *testing.T)`. You could also merge the `createTable` helper